### ISSUE TITLE
Engine should support reevaluation of formulas with no input columns

### ISSUE THREAD

**Issue by rbasralian** [deephaven/deephaven-core#4184]
Formulas that have no column inputs (like `.update("Timestamp = currentTime()")`) only tick on initialization/new rows. For this and similar formulas, it's usually desirable to be able to ignore MCS and force the formula to be evaluated any time the row is modified. DHC should have an equivalent to DHE's `.update(SelectColumnFactory.ofAlwaysUpdate("Timestamp = currentTime()"))` to work around this.

The closest way of emulating this right now would be to pass every column in the table as an input to a formula that returns a timestamp:

```groovy
myTimestampProducer = { -> return now() }
allColNamesStr = String.join(", ", t.getDefinition().getColumnNamesArray())
t2 = t.update("Timestamp = (Instant) myTimestampProducer($allColNamesStr)")
```

(but that still isn't quite the same thing, and it would be expensive because you'd have to read every column)

**Comment by cpwright** [comment]
fba05da70ae9d09a89299e7f58f7296e2dca170b
de72bf968cd978058dfb985b3dacd3dcbb4fdf49
23a91ab00f0cfae295b2de6fc760a7a184a9b228

**Comment by cpwright** [comment]
Fixed by #6339 .
